### PR TITLE
Adds package / release process + more

### PR DESCRIPTION
**Pull request by kfiku**
Adds tslint, adds seperation of renderer and main process, autosaving window boundries.
I also move electron.js to src/main/main.ts and add fuse build process for it (with quantum plugin).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,30 +1,92 @@
...
+    "url": "https://github.com/fuse-box"
+  },
+  "version": "1.0.5",
+  "description": "Electron application boilerplate based on typescript fusebox with hot reloading",
+  "main": "dist/main/main.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "node fuse",
+    "start:electron": "electron .",
+    "start:electron:watch": "nodemon --watch 'dist/main/*.*' --exec 'electron .'",
```
[the remark this answers is not in the document]
Comment (nchanged) [reply package.json]: And why do you need to restart electron process if there is a socket connection that does HMR?

[the remark this answers is not in the document]
Comment (kfiku) [reply package.json]: electron apps are build similar to back and front-end website. You have main process (like node app) and renderer (like front end). HMR are used in renderer in our case (so all files in src/renderer) but you can't hot reload backend electron process so you need restart app.

Here you have article on this topic: https://medium.com/@ccnokes/deep-dive-into-electrons-main-and-renderer-processes-7a9599d5c9e2

[the remark this answers is not in the document]
Comment (nchanged) [reply package.json]: Ok, lemme check

[the remark this answers is not in the document]
Comment (nchanged) [reply package.json]: I actually got an error 
`renderer.js:26 Uncaught SyntaxError: Unexpected token {`

The css plugin isn't working somehow

[the remark this answers is not in the document]
Comment (nchanged) [reply package.json]: rror: Node Sass does not yet support your current environment: OS X 64-bit with Unsupported runtime (57)
For more information on which environments are supported please see:
https://github.com/sass/node-sass/releases/tag/v4.5.2

[the remark this answers is not in the document]
Comment (nchanged) [reply package.json]: @kfiku ok, please upgrade node-sass there `"node-sass": "^4.5.2"`

[the remark this answers is not in the document]
Comment (kfiku) [reply package.json]: Ok. I will update also electron to latest...

[the remark this answers is not in the document]
Comment (nchanged) [reply package.json]: why not!

**Pull request merged**